lib/ukfile/include/uk/file/iovutil.h

```diff
@@ -0,0 +1,165 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2024, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+/* Convenience operations for dealing with iovec */
+
+#ifndef __UKFILE_IOVUTIL_H__
+#define __UKFILE_IOVUTIL_H__
+
+#include <string.h>
+#include <sys/uio.h>
+
+#include <uk/essentials.h>
+
+/**
+ * Zero out at most `len` bytes in memory regions described by `iov[iovcnt]`,
+ * starting at `*curp` offset from the buffer at `iov[*iovip]`.
+ *
+ * If the total remaining space in `iov` is less than `len`, exit early.
+ * After the call, `*iovip` and `*curp` are updated with the new positions.
+ *
+ * @return Number of bytes zeroed
+ */
+static inline
+size_t uk_iov_zero(const struct iovec *iov, int iovcnt, size_t len,
+		   int *iovip, size_t *curp)
+{
+	size_t ret = 0;
+	int i = *iovip;
+	size_t cur = *curp;
+
```

Comment:
I think it should be checked that `*iovip < iovcnt` and `*curp < iov[*iovip]`, for all functions in this file.

Reply:
You mean as in bounds checking? Yeah, good catch, will add assertions for that. I'm choosing assertions over defined checks because:
- caller must ensure indexes are placed appropriately before the call => it is illegal to supply out-of-bounds args and we must crash loudly if that happens
- this code is intended to be used in inner loops, and checks should be optimized away for performance

Reply:
I agree assertions are better, in this case out-of-bounds access should crash.